lib/dice/src/lib.rs

```diff
@@ -189,18 +189,13 @@ impl SeedBuf for CdiL1 {
 }
 
 impl CdiL1 {
-    pub fn new(cdi: &Cdi, tcb_tci: &[u8; SEED_LENGTH]) -> Self {
-        let mut okm = [0u8; SEED_LENGTH];
-        // TODO: Not sure FWID should be the salt here as we don't know much
-        // about its entropy content (hash of FW). The CDI should have good
-        // entropy though so maybe the CDI should be the salt and the FWID
-        // should be the IKM? Does it matter?
-        let hk = Hkdf::<Sha3_256>::new(Some(tcb_tci), cdi.as_bytes());
-        // No info provided to 'expand', see RFC 5869 §3.2.
-        // TODO: return error instead of expect
-        hk.expand(&[], &mut okm).expect("failed to expand");
-
-        CdiL1(okm)
+    pub fn new(cdi: &Cdi, tci: &[u8; SEED_LENGTH]) -> Self {
+        let mut hmac = Hmac::<Sha3_256>::new_from_slice(cdi.as_bytes())
+            .expect("HmacSha3_256");
```

Comment:
What would cause this condition to fail? It looks like sizeof(Cdi) being wrong?

I ask because `expect` (and `unwrap`) are both relatively expensive in terms of text size. Changing this to `unwrap` would improve things a bit. Using `unwrap_lite` would improve things a lot but that's only available in Hubris right now.